### ISSUE TITLE
findOrCreate with string intermittently not finding already-existing match

### ISSUE THREAD

**Issue by Ignigena** [balderdashy/sails#1825]
I am having a rather strange issue with the findOrCreate() function with the Mongo backend (apologies if this should be in the sails-mongo issue queue rather than this one)  Running the latest beta rc versions of sails, waterline and sails-mongo

The issue seems to be isolated to when using a string in the find field.  I have a function running on Cron to sync with an API and 99% of the time it works every time.  However, over the course of a few days, I'll start to notice duplicate entries sneaking in.  These are completely identical matches.  I've had to turn off the "unique" key in order to prevent this issue from randomly crashing the server.

Here's a super basic snippet with much of the meat cut out.  If I remove the match to the "uuid" field, which is defined in the model as a string, the issue appears to go away:

```
var videoId = video.uri.toString().replace('/videos/', '');
PodcastMedia.findOrCreate({
    uuid: videoId,
    podcast: new ObjectID(podcast.id)
  }, {
    name: video.name,
    date: new Date(video.created_time),
    description: video.description,
    tags: videoTags,
    duration: video.duration,
    thumbnail: videoThumbnail,
    url: videoUrl,
    uuid: videoId,
    podcast: new ObjectID(podcast.id)
  }, function podcastMediaCreated(err, media) {
    if (err) sails.log.error(err);
  });
```

The model:

```
    uuid: {
      type: 'string'
    },
    podcast: {
      model: 'podcast'
    }
```

And one of the duplicate entries created by this function:

```
{
  name: "Do You Want To Know A Secret?",
  date: ISODate("2011-05-12T03:10:07Z"),
  uuid: "23618802",
  podcast: ObjectId("53225a780c47fa1100b5ca3a"),
  createdAt: ISODate("2014-06-09T02:30:05.916Z"),
  updatedAt: ISODate("2014-06-09T02:30:05.916Z"),
  _id: ObjectId("53951c2d1081ab2000315872")
},
{
  name: "Do You Want To Know A Secret?",
  date: ISODate("2011-05-12T03:10:07Z"),
  uuid: "23618802",
  podcast: ObjectId("53225a780c47fa1100b5ca3a"),
  createdAt: ISODate("2014-06-09T00:30:06.067Z"),
  updatedAt: ISODate("2014-06-09T00:30:06.067Z"),
  _id: ObjectId("5395000e1081ab2000315805")
}
```

I haven't yet tried to replicate the issue with a different storage backend.

**Comment by mikermcneil** [comment]
Thanks for posting, @Ignigena.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
